### ISSUE TITLE
iOS: flutter build issue with multiple flavors

### ISSUE THREAD

**Issue by vladimirgoncharov** [flutter/flutter#88892]
Hello! I have an issue with multiple flavors. The issue is similar https://github.com/flutter/flutter/issues/18342 . 

I have the next configurations and schemes in iOS app:
<img width="157" alt="Screenshot 2021-08-25 at 21 53 40" src="https://user-images.githubusercontent.com/3529977/130841096-b1233f08-b192-4afd-9caf-73f528296869.png">
<img width="929" alt="Screenshot 2021-08-25 at 21 54 37" src="https://user-images.githubusercontent.com/3529977/130841192-79067cf4-0252-4250-9b6a-720a99568fb8.png">
```
xcrun xcodebuild -list
[+2008 ms] Command line invocation:
                        /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -list

                    User defaults from command line:
                        IDEPackageSupportUseBuiltinSCM = YES

                    Information about project "Runner":
                        Targets:
                            Runner

                        Build Configurations:
                            Debug-ccnc_prod
                            Debug-cn_dev
                            Debug-cn_prod
                            Debug-cn_uat
                            Debug-icp_prod
                            Debug-vh_dev
                            Profile-ccnc_prod
                            Profile-cn_dev
                            Profile-cn_prod
                            Profile-cn_uat
                            Profile-icp_prod
                            Profile-vh_dev
                            Release-ccnc_prod
                            Release-cn_dev
                            Release-cn_prod
                            Release-cn_uat
                            Release-icp_prod
                            Release-vh_dev

                        If no build configuration is specified and -scheme is not passed then "Debug-ccnc_prod" is used.

                        Schemes:
                            ccnc_prod
                            cn_dev
                            cn_prod
                            cn_uat
                            icp_prod
                            vh_dev
```
I tried to run `flutter build ios --flavor vh_dev --release --no-codesign --verbose` but flutter builds Debug-ccnc_prod configurations always.

I did a little research and found the issue in the command:
```
xcrun xcodebuild -project /Users/vladimir/Documents/@git/vh-flutter-heliosgo/ios/Runner.xcodeproj -scheme vh_dev -showBuildSettings
```
The command uses the default configuration.
Flutter should uses the next command for correct building:
```
xcrun xcodebuild -project /Users/vladimir/Documents/@git/vh-flutter-heliosgo/ios/Runner.xcodeproj -configuration Release-vh_dev -showBuildSettings
```
Could you fix the issue?

My local machine data:
```
flutter doctor -v
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.4 20F71 darwin-x64, locale en-RU)
    • Flutter version 2.2.3 at /usr/local/flutter
    • Framework revision f4abaa0735 (8 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/vladimir/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.59.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (2 available)
    • iPad Pro (12.9-inch) (5th generation) (mobile) • XXX • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)                                   • chrome                               • web-javascript • Google Chrome 92.0.4515.159

• No issues found!
```

**Comment by vladimirgoncharov** [comment]
@darshankawar I tried to build iOS app by the instruction https://flutter.dev/docs/deployment/cd#running-deployment-locally
1. ```
flutter build ios --release --no-codesign --flavor=flavor
build_app(workspace: XCODEWORKSPACE_PATH, scheme: flavor) // here the problem
upload_to_testflight
````

**Comment by darshankawar** [comment]
@vladimirgoncharov 
The instructions contains using `fastlane`. Do you see the same behavior without using it, ie manually too ?
For example, using this guide ? https://roszkowski.dev/2019/flutter-flavors/#flavors-on-ios

**Comment by vladimirgoncharov** [comment]
@darshankawar Yes. When I tried to build for custom flavor: 
```
flutter build ios --release --no-codesign --flavor=vh_dev
````
Flutter builds for default configuration ccnc_prod. Because flutter tried to build via command:
```
xcrun xcodebuild -project /Users/vladimir/Documents/@git/vh-flutter-heliosgo/ios/Runner.xcodeproj -scheme vh_dev -showBuildSettings
```
And xcrun said:
```
If no build configuration is specified and -scheme is not passed then "Debug-ccnc_prod" is used.
```
To solve this problem, flutter should run the command (via -configuration instead -scheme):
```
xcrun xcodebuild -project /Users/vladimir/Documents/@git/vh-flutter-heliosgo/ios/Runner.xcodeproj -configuration Release-vh_dev -showBuildSettings
``` 
This problem happens before `fastlane`.

**Comment by darshankawar** [comment]
Thanks for the details. I see the same behavior upon creating a sample flavors app.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[✓] Flutter (Channel stable, 2.2.3, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
    • Flutter version 2.2.3 at /Users/dhs/documents/fluttersdk/flutter
    • Framework revision f4abaa0735 (4 days ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/dhs/Library/Android/sdk
    • Java binary at: /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.    

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • SM G975F (mobile) • RZ8M802WY0X • android-arm64 • Android 10 (API 29)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 91.0.4472.114


• No issues found!

[✓] Flutter (Channel master, 2.6.0-1.0.pre.134, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.6.0-1.0.pre.134 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6902a04c22 (3 hours ago), 2021-08-30 23:01:05 -0400
    • Engine revision 6c11ffe9ba
    • Dart version 2.15.0 (build 2.15.0-66.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/dhs/Library/Android/sdk
    • Java binary at: /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.    

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • SM G975F (mobile) • RZ8M802WY0X • android-arm64 • Android 10 (API 29)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 91.0.4472.114



```

</details>

**Comment by jmagman** [comment]
This was fixed #80904, which is available on [2.3.0-1.0.pre](https://flutter.dev/docs/development/tools/sdk/releases?tab=macos) (currently the [beta channel](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels#how-to-change-channels)).
